niceman/resource/tests/test_ssh.py

```diff
@@ -17,11 +17,44 @@
 from ...utils import swallow_logs
 from ...tests.utils import assert_in, skip_if_no_network, skip_ssh
 from ..base import ResourceManager
+from ...cmd import Runner
 
+import pytest
+from nose import SkipTest
 
-@skip_ssh
-@skip_if_no_network
-def test_ssh_class():
+@pytest.fixture(scope='module')
+def setup_docker():
+    """pytest fixture for tests needing a running docker container
+
+    on setup, this fixture ensures that a docker container that maps 
+    host port 49000 to container port 22 is running and starts one if necessary
+
+    on teardown, this fixture stops the docker container if it was started by 
+    the fixture
+    """
+    skip_if_no_network()
+    stdout, _ = Runner().run(['docker', 'ps'])
+    if '0.0.0.0:49000->22/tcp' in stdout:
+        stop_container = False
+    else:
+        args = ['docker', 
+                'run', 
+                '-d', 
+                '--rm', 
+                '-p', 
+                '49000:22', 
+                'rastasheep/ubuntu-sshd:14.04']
+        stdout, _ = Runner().run(args)
+        container_id = stdout.strip()
+        stop_container = True
+    yield
+    if stop_container:
+        Runner().run(['docker', 'stop', container_id])
+    return
+
+def test_ssh_class(setup_docker):
+
+    skip_ssh()
```

Comment:
1. just move `skip_ssh` inside the `setup_docker`.  This way we wouldn't need to call it in every function which uses setup_docker
2. please rename setup_docker to `setup_ssh` (since it is not per se about (any) docker but rather about establishing ability to test ssh).  

And then it will be done (although ideally, the fixture should provide those parameters as host, user, password, port ;) but if you aren't up for this challenge -- we could do that later)

Reply:
These two options don't play well together.  It made more sense to me to make this a general docker fixture (see above), but it's dragging on a bit and at this point I'd rather just close this and move on to something more productive.  Let's discuss on Monday with Matt as well (who was working on the ssh bit).

Reply:
More productive is just to get it done... I could do it if you like - it is just few lines changes. but could you explain what you meant by "don't play well together"?

Reply:
Refactoring into a generic docker fixtures probably should be done whenever we have the use case for that. ATM it is just the SSH over